test/e2e/chaos/networkchaos/network_delay.go

```diff
@@ -257,7 +261,45 @@ func TestcaseNetworkDelay(
 	framework.ExpectNoError(err, "delete network chaos error")
 
 	wait.Poll(time.Second, 15*time.Second, func() (done bool, err error) {
-		result = probeNetworkCondition(c, networkPeers, ports)
+		result = probeNetworkCondition(c, networkPeers, ports, false)
+		if len(result[networkConditionBlocked]) != 0 || len(result[networkConditionSlow]) != 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	framework.ExpectEqual(len(result[networkConditionBlocked]), 0)
+	framework.ExpectEqual(len(result[networkConditionSlow]), 0)
+
+	bothDirectionNetem := makeNetworkDelayChaos(
+		ns, "network-chaos-4",
+		map[string]string{"app": "network-peer-0"},
+		map[string]string{"partition": "0"}, // 0 -> even its partition (idx % 2)
+		v1alpha1.OnePodMode,
+		v1alpha1.AllPodMode,
+		v1alpha1.Both,
+		testDelayTcParam,
+		testDelayDuration,
+		testDelaySchedulerSpec,
+	)
+	By("Injecting both direction chaos for 0")
+	err = cli.Create(ctx, bothDirectionNetem.DeepCopy())
+	framework.ExpectNoError(err, "create network chaos error")
+	wait.Poll(time.Second, 15*time.Second, func() (done bool, err error) {
+		result = probeNetworkCondition(c, networkPeers, ports, true)
+		if len(result[networkConditionBlocked]) != 0 || len(result[networkConditionSlow]) != 2 {
+			return false, nil
+		}
+		return true, nil
```

Comment:
How about inverting this `if` statement.

```go
		if len(result[networkConditionBlocked]) == 0 && len(result[networkConditionSlow]) == 2 {
			return true, nil
		}
		return false, nil
```

Reply:
I prefer to keep the same pattern with other test cases (as it's hard to notice the difference for the next time modifying the test case)

Reply:
Oh, I just realize it. 👍